homeassistant/components/reolink/config_flow.py

```diff
@@ -61,16 +70,37 @@ def async_get_options_flow(
         """Options callback for Reolink."""
         return ReolinkOptionsFlowHandler(config_entry)
 
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an authentication error or no admin privileges."""
+        self._host = entry_data[CONF_HOST]
+        self._username = entry_data[CONF_USERNAME]
+        self._password = entry_data[CONF_PASSWORD]
+        self._reauth = True
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        if user_input is not None:
+            return await self.async_step_user()
+        return self.async_show_form(step_id="reauth_confirm")
+
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle the initial step."""
         errors = {}
-        placeholders = {}
+        placeholders = {"error": ""}
 
         if user_input is not None:
+            host = ReolinkHost(self.hass, user_input, DEFAULT_OPTIONS)
             try:
-                host = await async_obtain_host_settings(self.hass, user_input)
+                await async_obtain_host_settings(host)
+            except UserNotAdmin:
+                errors[CONF_USERNAME] = "not_admin"
+                placeholders["username"] = host.api.username
```

Comment:
@starkillerOG I cherry-picked #85570 because it was tagged for a hot fix but am getting pylint errors. Is `rc` missing a version bump that contains this data? 

Can you link a PR to pick that fixes this or shall I remove this PR from the hot fix?

Reply:
I think it is this one https://github.com/home-assistant/core/pull/85848 but there is already a newer bump in dev https://github.com/home-assistant/core/pull/85871

Reply:
It's this one that's needed.

https://github.com/home-assistant/core/pull/85571

It bumps the library to 0.2.1.

Reply:
I came here with the same conclusion, needs #85571

Will pick it

Reply:
Thanks!

Reply:
@MartinHjelmare is right, it is https://github.com/home-assistant/core/pull/85571 bump reolink-aio to 0.2.1.